bin/check-ts.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Script to run the type-checker in all directories in the npm workspaces list
+
+set -e
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+REPO="$SCRIPT_DIR/.."
+
+cd "$REPO"
+DIRS=$( jq --raw-output ".workspaces[]" package.json )
+
+for dir in $DIRS
+do
+    "$SCRIPT_DIR"/cd_sh "$dir" "[ ! -f eslint.config.js ] || npx tsc"
```

Comment:
Could this be replaced with a simple `npm run build` in each directory?

Reply:
I don't know how to do that and still get the printing before and after. Am I missing something?

Reply:
No, I meant:
```suggestion
    "$SCRIPT_DIR"/cd_sh "$dir" "npm run build"
```

Reply:
Ah, ok. `npm run build` fails in the eslint-config and tsconfig directories. There is probably a way to change their package.json files to make it pass. Would that be better?

Reply:
Either that or do `npm run build --if-present` which succeeds when the `build` script is not defined.

Reply:
Ok the --if-present approach works, updated to do that. This does result in `make check-ts` actually build stuff with skargo, which I don't know if that is necessary for just checking the ts sources, but once it is done once it isn't significantly slower.